src/cpufreq.c

```diff
@@ -88,11 +118,44 @@ int cpufreq_init_cluster(const struct cluster_t *cluster, const struct feat_t *f
         }
     }
 
-    /* Unknown */
-    write64(cluster->base + 0x440f8, 1);
+    int pmgr_path[8];
+    u64 pmgr_reg;
+
+    if (adt_path_offset_trace(adt, "/arm-io/pmgr", pmgr_path) < 0) {
+        printf("Error getting /arm-io/pmgr node\n");
+        return -1;
+    }
+
+    if (adt_get_reg(adt, pmgr_path, "reg", 0, &pmgr_reg, NULL) < 0) {
+        printf("Error getting /arm-io/pmgr regs\n");
+        return -1;
+    }
+
+    switch (chip_id) {
+        case S5L8960X:
+            write32(pmgr_reg + PMGR_VOLTAGE_CTL_OFF_S5L8960X, 1);
+            break;
+        case T7000 ... T7001:
+            write32(pmgr_reg + PMGR_VOLTAGE_CTL_OFF_T7000, 1);
+            break;
+        case S8000 ... T8015:
+            /*
+             * On T8015 this will result in the register being written
+             * two times (for two clusters). However, this is fine.
+             */
+            write32(pmgr_reg + PMGR_VOLTAGE_CTL_OFF_S8000, 1);
+            break;
+        case T8103 ... T8122:
+        case T6000 ... T6034:
+        default:
```

Comment:
Please also separate out the default case and just print that the chip is unsupported